test/synapse/2019-06-01/azspark/zz_generated_batch_client.go

```diff
@@ -19,17 +20,25 @@ import (
 	"strings"
 )
 
-type batchClient struct {
+// BatchClient contains the methods for the SparkBatch group.
+// Don't use this type directly, use NewBatchClient() instead.
+type BatchClient struct {
 	endpoint string
 	pl       runtime.Pipeline
 }
 
-// newBatchClient creates a new instance of batchClient with the specified values.
+// NewBatchClient creates a new instance of BatchClient with the specified values.
 // endpoint - The workspace development endpoint, for example https://myworkspace.dev.azuresynapse.net.
 // livyAPIVersion - Valid api-version for the request.
 // sparkPoolName - Name of the spark pool.
-// pl - the pipeline used for sending requests and handling responses.
-func newBatchClient(endpoint string, livyAPIVersion *string, sparkPoolName string, pl runtime.Pipeline) *batchClient {
+// credential - used to authorize requests. Usually a credential from azidentity.
+// options - pass nil to accept the default values.
+func NewBatchClient(endpoint string, livyAPIVersion *string, sparkPoolName string, credential azcore.TokenCredential, options *BatchClientOptions) *BatchClient {
```

Comment:
This poses an interesting problem.  `livyAPIVersion` is optional so it belongs in the options type.  But we're not generating the options type.  We don't have to fix it in this PR, but we can't ship any packages like this.